demisto_sdk/commands/lint/helpers.py

```diff
@@ -0,0 +1,350 @@
+    """ Test internet connection
+
+    Args:
+        url(str): URL to test with.
+        timeout(int): timeout in seconds
+
+    Returns:
+        bool: True if internet connection availble, else False.
+    """
+    try:
+        _ = requests.get(url, timeout=timeout)
+        return True
+    except requests.ConnectionError:
+        print("No internet connection available.")
+    return False
+
+
+def build_skipped_exit_code(no_flake8: bool, no_bandit: bool, no_mypy: bool, no_pylint: bool, no_vulture: bool,
+                            no_test: bool, no_pwsh_analyze, no_pwsh_test, docker_engine: bool) -> bool:
+    """
+    no_flake8(bool): Whether to skip flake8.
+    no_bandit(bool): Whether to skip bandit.
+    no_mypy(bool): Whether to skip mypy.
+    no_vulture(bool): Whether to skip vulture
+    no_pylint(bool): Whether to skip pylint.
+    no_test(bool): Whether to skip pytest.
+    docker_engine(bool): docker engine exists.
+    """
+    skipped_code = 0b0
+    if no_flake8:
+        skipped_code |= EXIT_CODES["flake8"]
+    if no_bandit:
+        skipped_code |= EXIT_CODES["bandit"]
+    if no_mypy or not docker_engine:
+        skipped_code |= EXIT_CODES["mypy"]
+    if no_vulture or not docker_engine:
+        skipped_code |= EXIT_CODES["vulture"]
+    if no_pylint or not docker_engine:
+        skipped_code |= EXIT_CODES["pylint"]
+    if no_test or not docker_engine:
+        skipped_code |= EXIT_CODES["pytest"]
+    if no_pwsh_analyze or not docker_engine:
+        skipped_code |= EXIT_CODES["pwsh_analyze"]
+    if no_pwsh_test or not docker_engine:
+        skipped_code |= EXIT_CODES["pwsh_test"]
+
+    return skipped_code
+
+
+def get_test_modules(content_repo: Optional[git.Repo]) -> Dict[Path, bytes]:
+    """ Get required test modules from content repository - {remote}/master
+    1. Tests/demistomock/demistomock.py
+    2. Tests/scripts/dev_envs/pytest/conftest.py
+    3. Scripts/CommonServerPython/CommonServerPython.py
+    4. CommonServerUserPython.py
+
+    Returns:
+        dict: path and file content - see below modules dict
+    """
+    modules = [Path("Tests/demistomock/demistomock.py"),
+               Path("Tests/scripts/dev_envs/pytest/conftest.py"),
+               Path("Packs/Base/Scripts/CommonServerPython/CommonServerPython.py"),
+               Path("Tests/demistomock/demistomock.ps1"),
+               Path("Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1")]
+    modules_content = {}
+    if content_repo:
+        remote = content_repo.remote()
+        for module in modules:
+            modules_content[module] = content_repo.commit(f'{remote.name}/master').tree[
+                str(module)].data_stream.read()
+    else:
+        for module in modules:
+            url = f'https://raw.githubusercontent.com/demisto/content/master/{module}'
+            modules_content[module] = requests.get(url=url,
```

Comment:
doesn't look like you are checking for requests status code. For example you could get a 404 and not notice.

Reply:
Change it to be handled with res.ok

Reply:
Also add trials, so it will not fail on first trial